// @storybook/addon-vitest/dist/_node-chunks/vitest.config-KMFSIDBY.js
import CJS_COMPAT_NODE_URL_93yk2j72xw4 from 'node:url';
import CJS_COMPAT_NODE_PATH_93yk2j72xw4 from 'node:path';
import CJS_COMPAT_NODE_MODULE_93yk2j72xw4 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_93yk2j72xw4.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_93yk2j72xw4.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_93yk2j72xw4.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import "./chunk-SGRNOCCG.js";

// raw:../templates/vitest.config.template
var vitest_config_default = `import path from 'node:path';
import { fileURLToPath } from 'node:url';

import { defineConfig } from 'vitest/config';

import { storybookTest } from '@storybook/addon-vitest/vitest-plugin';

const dirname =
  typeof __dirname !== 'undefined' ? __dirname : path.dirname(fileURLToPath(import.meta.url));

// More info at: https://storybook.js.org/docs/next/writing-tests/integrations/vitest-addon
export default defineConfig({
  test: {
    workspace: [
      {
        extends: true,
        plugins: [
          // The plugin will run tests for the stories defined in your Storybook config
          // See options at: https://storybook.js.org/docs/next/writing-tests/integrations/vitest-addon#storybooktest
          storybookTest({ configDir: path.join(dirname, 'CONFIG_DIR') }),
        ],
        test: {
          name: 'storybook',
          browser: {
            enabled: true,
            headless: true,
            provider: 'playwright',
            instances: [{ browser: 'chromium' }],
          },
        },
      },
    ],
  },
});
`;
export {
  vitest_config_default as default
};
